Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveHM_2nd.cpp

```diff
@@ -510,18 +538,23 @@ void FiniteDifferenceSolver::MacroscopicEvolveHMCartesian_2nd(
                         if (mag_exchange_coupling == 1){
                             // H_exchange
                             if (mag_exchange_arrx == 0._rt) amrex::Abort("The mag_exchange_arrx is 0.0 while including the exchange coupling term H_exchange for H_eff");
+
+                            LapM_prev_xface(i, j, k, 0) = T_Algo::Laplacian(M_prev_xface, coefs_x, coefs_y, coefs_z, i, j, k, 0); // M_xface from previous iteration of current time step
+                            LapM_prev_xface(i, j, k, 1) = T_Algo::Laplacian(M_prev_xface, coefs_x, coefs_y, coefs_z, i, j, k, 1); // M_xface from previous iteration of current time step
+                            LapM_prev_xface(i, j, k, 2) = T_Algo::Laplacian(M_prev_xface, coefs_x, coefs_y, coefs_z, i, j, k, 2); // M_xface from previous iteration of current time step
+
                             amrex::Real const H_exchange_coeff = 2.0 * mag_exchange_arrx / PhysConst::mu0 / mag_Ms_arrx / mag_Ms_arrx;
-                            Hx_eff += H_exchange_coeff * T_Algo::Laplacian(M_xface, coefs_x, coefs_y, coefs_z, i, j, k, 0);
-                            Hy_eff += H_exchange_coeff * T_Algo::Laplacian(M_xface, coefs_x, coefs_y, coefs_z, i, j, k, 1);
-                            Hz_eff += H_exchange_coeff * T_Algo::Laplacian(M_xface, coefs_x, coefs_y, coefs_z, i, j, k, 2);
+                            Hx_eff += H_exchange_coeff * LapM_prev_xface(i, j, k, 0);
+                            Hy_eff += H_exchange_coeff * LapM_prev_xface(i, j, k, 1);
+                            Hz_eff += H_exchange_coeff * LapM_prev_xface(i, j, k, 2);
```

Comment:
same here. Just use `M_prev`

Reply:
How about
x_eff += H_exchange_coeff * 0.5 * (LapM_prev_xface(i, j, k, 0) + LapM_old_xface(i,j,k,0);
you can additionally swap in the call to T_Algo::Laplacian(M_prev_xface) here to save memory